infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/provision/OpenShiftStopWorkspaceRoleProvisioner.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.openshift.provision;
+
+import com.google.inject.Inject;
+import io.fabric8.kubernetes.api.model.ObjectReferenceBuilder;
+import io.fabric8.openshift.api.model.*;
+import io.fabric8.openshift.client.OpenShiftClient;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.workspace.infrastructure.openshift.OpenShiftClientFactory;
+import org.eclipse.che.workspace.infrastructure.openshift.environment.OpenShiftCheInstallationLocation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class creates the necessary role and rolebindings to allow the che serviceaccount to stop
+ * user workspaces.
+ *
+ * @author Tom George
+ */
+public class OpenShiftStopWorkspaceRoleProvisioner {
+
+  private final OpenShiftClientFactory clientFactory;
+  private final OpenShiftCheInstallationLocation installationLocation;
+
+  private static final Logger LOG = LoggerFactory.getLogger(OpenShiftCheInstallationLocation.class);
+
+  @Inject
+  public OpenShiftStopWorkspaceRoleProvisioner(
+      OpenShiftClientFactory clientFactory, OpenShiftCheInstallationLocation installationLocation) {
+    this.clientFactory = clientFactory;
+    this.installationLocation = installationLocation;
```

Comment:
I think it makes more sense to inject `OpenShiftCheInstallationLocation` but store `OpenShiftCheInstallationLocation.getInstallationLocationNamespace()` as a String in the class. It shouldn't change while Che is running, right?